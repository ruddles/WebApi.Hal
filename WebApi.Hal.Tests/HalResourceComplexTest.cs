using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using WebApi.Hal.Tests.Representations;
using Xunit;

namespace WebApi.Hal.Tests
{
    public class HalResourceComplexTest
    {
        readonly OrganisationsRepresentation organisations;

        public HalResourceComplexTest()
        {
            organisations = new OrganisationsRepresentation
            {
                Organisations = new Collection<OrganisationWithPeopleRepresentation>
                {
                    new OrganisationWithPeopleRepresentation(1, "first"),
                    new OrganisationWithPeopleRepresentation(2, "second")
                }
            };
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void children_have_self_link_test()
        {
            // arrange
            var mediaFormatter = new JsonHalMediaTypeFormatter { Indent = true };
            var content = new StringContent(string.Empty);
            var type = organisations.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, organisations, stream, content, null).Wait();
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                Console.WriteLine(serialisedResult);
                Approvals.Verify(serialisedResult);
            }
        }
    }
}
