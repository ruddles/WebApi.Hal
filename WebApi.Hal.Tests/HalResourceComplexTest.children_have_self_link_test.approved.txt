{
  "Organisations": [
    {
      "Id": 1,
      "Name": "first",
      "_links": {
        "self": {
          "href": "/api/organisations/1"
        },
        "people": {
          "href": "/api/organisations/1/people"
        }
      }
    },
    {
      "Id": 2,
      "Name": "second",
      "_links": {
        "self": {
          "href": "/api/organisations/2"
        },
        "people": {
          "href": "/api/organisations/2/people"
        }
      }
    }
  ],
  "_links": {
    "self": {
      "href": "/organisations"
    }
  }
}